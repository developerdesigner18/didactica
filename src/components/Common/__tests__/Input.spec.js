import { createLocalVue, mount } from "@vue/test-utils";
import { waitRAF } from "@/../tests/utils.js";

import LxpInput from "../Input.vue";

// create an extended `Vue` constructor
const localVue = createLocalVue();

localVue.component("lxp-input", LxpInput);

describe("Input.vue", () => {
  it("render when created ", async () => {
    let wrapper = await getWrapper();
    expect(wrapper.exists()).toBeTruthy();
    expect(wrapper.element).toMatchSnapshot();
    wrapper.destroy();
  });
  describe("Props", () => {
    let wrapper;

    afterEach(() => {
      wrapper.destroy();
    });

    it("should show text limit and supporting text", async () => {
      wrapper = await getWrapper();
      await wrapper.setData({ text: "Hello" });
      await wrapper.setProps({
        maxlength: 100,
        showLimit: true,
        supportText: "A supporting text"
      });
      await waitRAF();

      const inputLimit = wrapper.find(".lxp-input__footer--limit");
      expect(inputLimit.exists()).toBe(true);

      const inputSupportText = wrapper.find(".lxp-input__footer--support-text");
      expect(inputSupportText.text()).toEqual("A supporting text");
    });

    it("should show border when `hasError` is set to be true ", async () => {
      wrapper = await getWrapper();
      await wrapper.setProps({
        hasError: true
      });
      await waitRAF();
      const inputLimit = wrapper.find(".lxp-input__error");
      expect(inputLimit.exists()).toBe(true);
    });
  });
});

function getWrapper() {
  return mount(LxpInput, {
    propsData: {
      label: "Input label",
      placeholder: "Placeholder"
    },
    localVue
  });
}
