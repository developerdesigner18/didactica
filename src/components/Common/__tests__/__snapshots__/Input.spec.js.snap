// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input.vue render when created  1`] = `
<div
  class="lxp-input"
>
  <div
    class="form-floating"
  >
    <!---->
     
    <div
      class="form-floating__div"
    >
      <input
        class="form-control"
        id="floatingInput"
        placeholder="Placeholder"
        type="text"
      />
       
      <label
        for="floatingInput"
      >
        Input label
      </label>
    </div>
     
    <!---->
  </div>
   
  <div
    class="lxp-input__footer"
  >
    <!---->
     
    <!---->
  </div>
</div>
`;
